rc/lsp.kak

```diff
@@ -342,9 +342,39 @@ have_kakoune_feature_filtertext = ${kak_opt_lsp_have_kakoune_feature_filtertext}
 " | eval "${kak_opt_lsp_cmd} --request") > /dev/null 2>&1 < /dev/null & }
 }}
 
+declare-option -hidden str lsp_completion_selection ""
+
 define-command -hidden lsp-completion-dismissed -docstring "Called when the completion pager is closed" %{
+    set-option window lsp_completion_selection %val{hook_param}
+    try %{
+        eval -draft %{
+            select %opt{lsp_completion_selection}
+        }
+    } catch %{
+        # Completion was cancelled, nothing was inserted
+        # Clear these so the following lines do nothing.
+        # This is to avoid runing them in the above try block, and silencing their errors
+        remove-hooks window lsp-completion-dismissed
+        set-option window lsp_completions_selected_item -1
+    }
     lsp-completion-item-resolve-request false
```

Comment:
I think this can be moved inside the `try` block since in the failure cause the selected item is always -1 (I hope).
Also the next line that triggers the user hook because we don't need to run insert snippets if the above `select` fails right?

Reply:
So this seems to work fine for me:

	define-command -hidden lsp-completion-dismissed -docstring "Called when the completion pager is closed" %{
	    set-option window lsp_completion_selection %val{hook_param}
	    try %{
	        eval -draft %{
	            select %opt{lsp_completion_selection}
	        }
	        lsp-completion-item-resolve-request false
	        trigger-user-hook "LspCompletionDismissed"
	    }
	    remove-hooks window lsp-completion-dismissed
	    set-option window lsp_completions_selected_item -1
	    set-option window lsp_completion_selection ""
	}

Reply:
I wanted to avoid this, because it means errors in the user hook are silenced

Reply:
Of course, sorry I missed that comment